### PR TITLE
Implement RCons.setTitle() for Windows and add ?et command ##cons

### PR DESCRIPTION
**Pull request by trufae**
DO NOT SQUASH

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

there's a w32api for that

**Test plan**

requires humans

**Closing issues**

none registered, this api was not used anywhere

### CODE REVIEW

File: libr/core/cmd_help.c [hunk 1]
```diff
@@ -949,6 +949,9 @@ static int cmd_help(void *data, const char *input) {
 		break;
 	case 'e': // "?e" echo
 		switch (input[1]) {
+		case 't': // "?e=t newtitle"
```
Comment (ret2libc) [inline libr/core/cmd_help.c:952]:
```suggestion
		case 't': // "?et newtitle"
```

**Review by ret2libc — APPROVED** [review]

**Review by wargio — APPROVED** [review]

File: libr/cons/cons.c [hunk 2]
```diff
@@ -1677,7 +1677,11 @@ R_API void r_cons_set_last_interactive(void) {
 }
 
 R_API void r_cons_set_title(const char *str) {
+#if __WINDOWS__
+	SetConsoleTitle (str);
```
Comment (a1ext) [inline libr/cons/cons.c:1681]: You have to use "wide" version of this API and you need to convert the input string. Actually it displays garbage (treats str as `wchar_t*`:
image

**Review by a1ext — CHANGES_REQUESTED** [review]

File: libr/cons/cons.c (hunk 2 above)
Comment (a1ext) [reply libr/cons/cons.c:1681]: I suggest it should look like:
```
#if __WINDOWS__
#  if defined(_UNICODE)
wchar_t* wstr = r_utf8_to_utf16_l(str, strlen(str));
if (wstr) {
    SetConsoleTitleW (wstr);
    R_FREE(wstr);
}
#  else // defined(_UNICODE)
SetConsoleTitle (str);
# endif
#endif // __WINDOWS__
```

Comment (trufae) [reply libr/cons/cons.c:1688]: applied your patch

Comment (a1ext) [reply libr/cons/cons.c:1688]: I have compiled your recent commit with the following meson command to turn on the unicode support:
`python sys\meson.py --shared --install --options "c_args=-D_UNICODE -DUNICODE"`

and it works well with utf8 console charset in default Windows terminal:
image

**Review by a1ext — APPROVED** [review]
👍

**Review by XVilka — APPROVED** [review]

**Pull request merged**